src/darwin/Framework/CHIP/MTRUtilities.mm

```diff
@@ -19,5 +19,16 @@
 
 BOOL MTREqualObjects(id<NSObject> _Nullable a, id<NSObject> _Nullable b)
 {
-    return (a == nil) ? (b == nil) : [a isEqual:b];
+    // Check if A is nil, and return based on B being nil (or not)
```

Comment:
So here's what confuses me.  The new logic here is equivalent to the old logic, so that's fine as far as it goes.  But why did we need to change this at all?  isEqual is documented as:

```
object

    The object to be compared to the receiver. May be nil, in which case this method returns NO.
```

so if `b` is nil and we reached the `[a isEqual:b]` stuff we would return NO as desired.  The new code is more explicit about it, I guess?

Reply:
Yes it is, and readable.

Reply:
The old code allowed you to pass nil to [a isEqual: b];